nrpy/infrastructures/BHaH/CurviBoundaryConditions/CurviBoundaryConditions.py

```diff
@@ -87,7 +87,15 @@ def parity_conditions_symbolic_dot_products(
 Documented in: Tutorial-Start_to_Finish-Curvilinear_BCs.ipynb
 */
 """
-    return outstr + ccg.c_codegen(parity, lhs_strings, fp_type=fp_type)
+    calculation = ccg.c_codegen(
+        parity, lhs_strings, fp_type="double" if fp_type == "float" else fp_type
+    )
```

Comment:
).replace("REAL ", double_str)

Reply:
I have made this change for the replace and want to highlight that `fp_type="double" if fp_type == "float" else fp_type` is still needed to ensure the codegen uses the correct strong typing